Src/Newtonsoft.Json/Serialization/DefaultContractResolver.cs

```diff
@@ -1353,7 +1353,7 @@ protected virtual IList<JsonProperty> CreateProperties(Type type, MemberSerializ
             return orderedProperties;
         }
 
-        internal virtual PropertyNameTable GetNameTable()
+        internal virtual DefaultNameTable GetNameTable()
```

Comment:
Could this return `NameTable`?

Reply:
Unfortunately DefaultContractResolver uses "nameTable.Add(property.PropertyName)" which is implementation specific and not part of abstract class definition. We could add missing method to abstract class, but it won't be used, because DefaultContractResolver only used together with DefaultNameTable. Client who uses custom contract resolver could use derived version of NameTable as well (similar to how DefaultContractResolver handling this). So looks like polluting abstract class is not necessary. Please let me know if you have other suggestions.